tox.ini

```diff
@@ -42,9 +42,8 @@ commands = pre-commit run --all-files
 [testenv:lint]
 commands =
     pflake8
-    pylint {toxinidir}/src/
-    pylint {toxinidir}/tests/ --disable=E0401,W0212,W0621
```

Comment:
Removing this line. It will conflict with the change in 
https://github.com/canonical/charm-prometheus-juju-exporter/pull/34/files

One of the PR have to change. I guess

Reply:
I think it would be better to remove the edit in https://github.com/canonical/charm-prometheus-juju-exporter/pull/34/files. The idea is to use `tox.ini` from the template as it is and only changes `pyproject.yaml` when needed. And since in the template, `tests/` folder are excluded from lint test, we wouldn't need to disable checks. I will discuss w/ Martin for a decision.

Reply:
Rebased and conflicts resolved :)

Reply:
I agree that running `pylint` and `mypy` could be beneficial to be use in functional tests, but in unit tests it's not so beneficial and you will need to ignore lot of things.
We can probably do some tests and play with the exclusions in pyproject.toml so that mypy and pylint work for `tests` directory, but we will also need a document on how to use it because it will end up with everyone just giving ignore (pylint: ingore) everywhere.  What do you think about it?

Reply:
I don't think adding complexity by circumventing pyproject for test linting is worth it. If there's no way to configure per-directory rules in `pyproject.yaml` we should just skip mypy and pylint for tests. We still have black/isort/pep8.

Reply:
I think this conversation can be resolved to not block merging.